packages/backend/src/controllers/bills/bills.controller.ts

```diff
@@ -0,0 +1,191 @@
+    private readonly userStoreService: UserStoreService,
+    private readonly billUtil: BillUtil,
+  ) {}
+
+  @Get()
+  @ApiOperation({ summary: 'Retrieve all bills in the users flat.' })
+  @ApiOkResponse({
+    description: 'Bills retrieved successfully',
+    type: String,
+  })
+  async getBills(@User() user: DecodedIdToken): Promise<BillsResponseDto> {
+    const houseID = (await this.userStoreService.findOneByFirebaseId(user.uid))
+      .house;
+
+    return {
+      bills: await Promise.all(
+        (
+          await this.billStoreService.findAllForHouse(houseID)
+        ).map((bill) => {
+          return this.billUtil.covertBillDocumentToResponseDTO(
+            bill,
+            this.userStoreService,
+          );
+        }),
+      ),
+    };
+  }
+
+  @Post()
+  @ApiOperation({ summary: 'Create a new bill resource' })
+  @ApiCreatedResponse({
+    description: 'Bill created successfully',
+    type: BillResponseDto,
+  })
+  async createBill(
+    @Body() createBillDto: CreateBillDto,
+    @User() user: DecodedIdToken,
+  ): Promise<BillResponseDto> {
+    const owner = await this.userStoreService.findOneByFirebaseId(user.uid);
+    const users = await Promise.all(
+      createBillDto.users.map(async (u) => {
+        return {
+          id: (await this.userStoreService.findOneByFirebaseId(user.uid))._id,
+          amount: u.amount,
+          paid: u.paid,
+          proof: u.proof,
+        };
+      }),
+    );
+    const billModel: BillModel = {
+      name: createBillDto.name,
+      description: createBillDto.description,
+      house: owner.house,
+      owner: owner._id,
+      due: new Date(createBillDto.due),
+      users: users,
+    };
+    const bill = await this.billStoreService.create(billModel);
+    return {
+      name: bill.name,
+      description: bill.description,
+      owner: owner.firebaseId,
+      due: createBillDto.due,
+      users: createBillDto.users,
+    };
+  }
+
+  @Put(':id')
+  @ApiOperation({ summary: 'Update a bill.' })
+  @ApiOkResponse({
+    description: 'Bill updated successfully',
+    type: BillResponseDto,
+  })
+  @ApiUnauthorizedResponse({ description: 'not the bill owner' })
+  async updateBill(
+    @Param() idObject: string,
+    @Body() updateBillDto: UpdateBillDto,
+    @User() user: DecodedIdToken,
+  ): Promise<BillResponseDto> {
+    const billId = idObject['id'];
+    const bill = await this.billStoreService.findOne(billId);
+    const userObject = await this.userStoreService.findOneByFirebaseId(
+      user.uid,
+    );
+
+    if (bill.owner.valueOf() === userObject._id.valueOf()) {
+      const billDocument = await this.billStoreService.update(
+        bill._id,
+        updateBillDto,
+      );
+      return this.billUtil.covertBillDocumentToResponseDTO(
+        billDocument,
+        this.userStoreService,
+      );
+    } else
+      throw new HttpException('not the bill owner', HttpStatus.UNAUTHORIZED);
```

Comment:
Return Forbidden instead. Also, I would suggest inverting the if condition and throwing early